src/checkbox/checkbox.js

```diff
@@ -15,255 +15,148 @@
  * limitations under the License.
  */
 
-(function() {
-  'use strict';
+// Numeric constants used in this component.
+const numbers_ = Object.freeze({
+  TINY_TIMEOUT: 0.001
+});
+
+// CSS classes used in this component.
+const classes_ = Object.freeze({
```

Comment:
Copies are a necessary evil here. This is small enough no drawbacks should be seen in reasonable applications.

At least in V8, doing `this.classes = Object.freeze({});` in the constructor so all objects are constructed with the same properties, will let it take advantage of shadow classes (or whatever they are called) underneath. Keeping good performance.

Just assign in constructor and things will be fine.